forms/core/Application.php

```diff
@@ -1468,4 +1398,34 @@ public static function encode($string){
         return stripslashes(htmlspecialchars($string));
     }
 
+    public static function refreshTokenIfNeeded() {
+		$auth = array(
+			'access_token' => Settings::get('access_token'),
+			'refresh_token' => Settings::get('refresh_token')
+		);
+
+		if (!empty($auth['access_token']) && Settings::get('expiry') < time()) {

```

Comment:
Just confirming that we've tested in getting refreshed token after it pass the expire time possess no issue (or do we want to renew prior to the access_token expired)?

Reply:
I have not as I'm not sure how to manually expire a refresh token in the backend. If the API docs are to be believed then it should be fine though 🤔